src/main/java/com/amazonaws/encryptionsdk/model/KeyBlob.java

```diff
@@ -98,7 +98,11 @@ public KeyBlob(final EncryptedDataKey edk) {
      *             length.
      */
     private int parseKeyProviderIdLen(final byte[] b, final int off) throws ParseException {
-        keyProviderIdLen_ = PrimitivesParser.parseShort(b, off);
+        short len = PrimitivesParser.parseShort(b, off);
```

Comment:
These methods are all doing the same thing, just with different error messages. This should be centralized into a single helper method that all three use.

Reply:
Will do